File: C++/Check if all leaves are at same level.cpp
class Solution{
    int maxi,mini;
    void checkHeight(Node* root, int i){
        if(!root) return;
        if(!root->left && !root->right){
            maxi = max(maxi, i);
            mini = min(mini, i);
        }
        checkHeight(root->left, i+1);
        checkHeight(root->right, i+1);
    }
  public:
    bool check(Node *root)
    {
        if(!root) return true;
        maxi = 0, mini = INT_MAX;
        checkHeight(root,1);
        return maxi == mini;
    }
};

File: JAVA/Check if all leaves are at same level.java
class Solution
{
    boolean check(Node root)
    {
       Queue<Node> q=new LinkedList<>();
        q.add(root);
        q.add(null);
        int l=0;
        int p=0;
        while(!q.isEmpty()){
            Node k=q.poll();
            if(k==null){
                if(q.isEmpty()){
                    break;
                }else{
                    q.add(null);
                    l+=1;
                   continue; 
                }
            }
            if(k.left!=null){
                q.add(k.left);
            }
            if(k.right!=null){
                q.add(k.right);
            }
            if(k.left==null&&k.right==null){
                if(p!=0&&p!=l){
                    return false;
                }else{
                    p=l;
                }
            }
        }
        return true;
    }
}
